app/instance-initializers/key-responder.js

```diff
@@ -0,0 +1,51 @@
+import Ember from 'ember';
+
+export default {
+  name: 'ember-key-responder-instance',
+
+  initialize: function(instance) {
+
+    //TextField/TextArea are currently uninjectable, so we're going to hack our
+    //way in
+    Ember.TextSupport.reopen({
+      keyResponder: Ember.computed(function() {
+        return this.container.lookup('key-responder:main');
+      }).readOnly()
+    });
+
+    // Set up a handler on the document for keyboard events that are not
+    // handled by Ember's event dispatcher.
+    Ember.$(document).on('keyup.outside_ember_event_delegation', null,
+                         function(event) {
```

Comment:
I don't believe they're ever torn down. What's the best practice for an UNinitializer